client/store.go

```diff
@@ -4,18 +4,34 @@ import (
 	"context"
 
 	"github.com/libp2p/go-libp2p-core/routing"
+	record "github.com/libp2p/go-libp2p-record"
 )
 
 var _ routing.ValueStore = &Client{}
 
 // PutValue adds value corresponding to given Key.
 func (c *Client) PutValue(ctx context.Context, key string, val []byte, opts ...routing.Option) error {
-	return c.PutIPNS(ctx, []byte(key), val)
+	_, path, err := record.SplitKey(key)
+	if err != nil {
+		return err
+	}
+
+	return c.PutIPNS(ctx, []byte(path), val)
```

Comment:
We should have a check for if the key is a valid IPNS identifier as in https://github.com/ipfs/go-ipns/blob/f6e884731b943682a432e4719696c21d8f354b9c/record.go#L48-L49.

Whether this lives in the valuestore wrapper or lower in the client seems debatable, either way is fine by me. You can reuse the error if you want or not, doesn't seem a huge deal either way.